internal/execute/table/builder_cache.go

```diff
@@ -21,6 +21,34 @@ type Builder interface {
 	Release()
 }
 
+// KeyLookup is an interface for storing and retrieving
+// items by their group key.
```

Comment:
Grateful for this interface and the docs in it. Coulda used it before I started drawing things out yesterday.